shared/model.py

```diff
@@ -286,14 +287,55 @@ def filter_records(build, classification, severity, machine_id=None, os_name=Non
     @staticmethod
     def delete_records():
         try:
-            sec_weeks = MAX_WEEK_KEEP_RECORDS * 7 * 24 * 60 * 60
-            current_time = time()
-            time_weeks_ago = current_time - sec_weeks
-            q = db.session.query(Record).filter(Record.tsp_server < time_weeks_ago)
-            count = q.delete(synchronize_session=False)
+            for field in PURGE_FILTERED_RECORDS.keys():
+                if field == 'severity':
+                    for severity in PURGE_FILTERED_RECORDS[field].keys():
+                        if PURGE_FILTERED_RECORDS[field][severity]:
+                            age = time() - PURGE_FILTERED_RECORDS[field][severity] * 24 * 60 * 60
+                            q = db.session.query(Record)
+                            q = q.filter(Record.severity == severity)
+                            q = q.filter(Record.tsp_server < age)
+                            count = q.delete(synchronize_session=False)
+                            print("Deleted {} {} records".format(count, severity))
+                elif field == 'classification':
+                    for classification in PURGE_FILTERED_RECORDS[field].keys():
+                        if PURGE_FILTERED_RECORDS[field][classification]:
+                            age = time() - PURGE_FILTERED_RECORDS[field][classification] * 24 * 60 * 60
+                            q = db.session.query(Record.id).join(Record.classification)
+                            q = q.filter(Classification.classification.like(classification))
+                            q = q.filter(Record.tsp_server < age)
+                            count = db.session.query(Record).filter(Record.id.in_(q)).delete(synchronize_session=False)
+                            print("Deleted {} {} records".format(count, classification))
+                elif field == 'machine_id':
+                    for machine_id in PURGE_FILTERED_RECORDS[field].keys():
+                        if PURGE_FILTERED_RECORDS[field][machine_id]:
+                            age = time() - PURGE_FILTERED_RECORDS[field][machine_id] * 24 * 60 * 60
+                            q = db.session.query(Record)
+                            q = q.filter(Record.machine_id == machine_id)
+                            q = q.filter(Record.tsp_server < age)
+                            count = q.delete(synchronize_session=False)
+                            print("Deleted {} {} records".format(count, machine_id))
```

Comment:
this if ... block could be refactor for readability.  80% of the code in each block is repeated, a function that takes field as an argument can replace the entire if sections.

i.e.
```
purge_field(field):
    for field_category in PURGE_FILTERED_RECORDS[field].keys():
        if PURGE_FILTERED_RECORDS[field][field_category]:
            age = time() - PURGE_FILTERED_RECORDS[field][field_category] * 24 * 60 * 60
            if field_category == 'severity':
                # Special case for severity
            elif field_category == 'classification':
                # Special case for classification
            ....

            print("Deleted {} {} records".format(count, field_category))
```


The for blok becomes:

```
for field in PURGE_FILTERED_RECORDS.keys():
    purge_field(field)

```

for

Reply:
Good idea Alex. I've already done that, see my changes.